### PR TITLE
BAVL-97 changes to model to support location keys (and not IDs as originally thought). Some DB tweaks also.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
Minor tidy ups, nothing major (expanded a little some class renames).

Changes around location identifiers, date and times of appointments.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
...
 [![CircleCI](https://dl.circleci.com/status-badge/img/gh/ministryofjustice/hmpps-book-a-video-link-api/tree/main.svg?style=svg)](https://dl.circleci.com/status-badge/redirect/gh/ministryofjustice/hmpps-activities-management-api/tree/main)
 [![Docker Repository on Quay](https://quay.io/repository/hmpps/hmpps-book-a-video-link-api/status "Docker Repository on Quay")](https://quay.io/repository/hmpps/hmpps-book-a-video-link-api)
-[![API docs](https://img.shields.io/badge/API_docs-view-85EA2D.svg?logo=swagger)](https://book-a-video-link-dev.prison.service.justice.gov.uk/swagger-ui/index.html?configUrl=/v3/api-docs/swagger-config)
+[![API docs](https://img.shields.io/badge/API_docs-view-85EA2D.svg?logo=swagger)](https://book-a-video-link-api-dev.prison.service.justice.gov.uk/swagger-ui/index.html#/)
```
Comment (matt-crowson-moj) [inline README.md:5]: Fixes the Swagger link in the readme.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/locationsinsideprison/LocationsInsidePrisonClient.kt [hunk 2]
```diff
@@ -4,12 +4,14 @@ import org.springframework.web.reactive.function.client.WebClient
...
-import java.util.UUID
 
 class LocationsInsidePrisonClient(private val locationsInsidePrisonApiWebClient: WebClient) {
 
-  fun getLocation(locationId: UUID): Location? = locationsInsidePrisonApiWebClient.get()
-    .uri("/locations/{id}", locationId)
+  fun getLocationsByKeys(keys: List<String>): List<Location> = keys.mapNotNull(::getLocationByKey)
+
+  @Deprecated(message = "We are waiting on a post endpoint to avoid multiple location API calls")
+  private fun getLocationByKey(key: String): Location? = locationsInsidePrisonApiWebClient.get()
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/locationsinsideprison/LocationsInsidePrisonClient.kt:13]: Temporary until work by the locations adds an endpoint to support this.  Note this work has been ticketed up by them.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/locationsinsideprison/LocationsInsidePrisonClient.kt [hunk 3]
```diff
@@ -4,12 +4,14 @@ import org.springframework.web.reactive.function.client.WebClient
 import org.springframework.web.reactive.function.client.WebClientResponseException
 import reactor.core.publisher.Mono
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.client.locationsinsideprison.model.Location
-import java.util.UUID
 
 class LocationsInsidePrisonClient(private val locationsInsidePrisonApiWebClient: WebClient) {
 
-  fun getLocation(locationId: UUID): Location? = locationsInsidePrisonApiWebClient.get()
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/locationsinsideprison/LocationsInsidePrisonClient.kt:11]: We won't be using location ID's (as these could potentially change).  We will be using a business key instead which will remain constant.

File: src/main/resources/migrations/common/V2024.04.22__baseline_tables.sql [hunk 4]
```diff
@@ -207,9 +207,10 @@ CREATE TABLE prison_appointment
     prisoner_number        varchar(7) NOT NULL,  -- NOMS number
     appointment_type       varchar(40) NOT NULL,
     comments               varchar(200),
-    prison_loc_id          bigint NOT NULL,  -- from locations API
-    start_time             timestamp NOT NULL,
-    end_time               timestamp NOT NULL,
+    prison_loc_key         varchar(160) NOT NULL,  -- from locations API
+    on_day                 date NOT NULL,
```
Comment (matt-crowson-moj) [inline src/main/resources/migrations/common/V2024.04.22__baseline_tables.sql:211]: Note sure on the name.  Could just say start_date?

Comment (timharrison-moj) [reply src/main/resources/migrations/common/V2024.04.22__baseline_tables.sql:211]: `appointment_date` ?

Comment (matt-crowson-moj) [reply src/main/resources/migrations/common/V2024.04.22__baseline_tables.sql]: Done

**Review by timharrison-moj — APPROVED** [review]
LGTM

**Pull request merged**